src/lib/sql/sql_translator.cpp

```diff
@@ -654,6 +657,32 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_stored_table(
   return validated_stored_table_node;
 }
 
+std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_meta_table(
+    const std::string& name, const std::shared_ptr<SQLIdentifierResolver>& sql_identifier_resolver) {
+  AssertInput(Hyrise::get().storage_manager.has_table(name), std::string{"Did not find a table with name "} + name);
+  AssertInput(MetaTableManager::is_meta_table_name(name), std::string{"Did not find a meta table with name "} + name);
+
+  _cacheable = false;
+
+  const auto meta_table = Hyrise::get().storage_manager.get_table(name);
+
+  const auto static_table_node = StaticTableNode::make(meta_table);
```

Comment:
Should we have a flag in StoredTableNode that prevents deep_copy from being called on meta tables?

Reply:
do you mean StaticTableNode? what's exactly the problem with deep copy here?

Reply:
deep_copy is called when a table is reused. Those table should not be reused, but recreated, so this would be a sanity check.

Reply:
i don't get it yet. isn't a table noncopyable?

Reply:
Yes, but deep-copying the StaticTableNode only copies the pointer to the table. As such, while the table is not copied, it is erroneously reused.

Reply:
i see. maybe a similar but different problem that came to my mind: how do we deal with static table nodes in prepared statements? they might get outdated at the time of execution.

Reply:
Good that you think of it. I don't think that we lose anything if we simply prohibit the combination of meta tables and prepared statements.